src/Microsoft.Data.SqlClient/tests/ManualTests/makepfxcert.ps1

```diff
@@ -0,0 +1,140 @@
+    $fqdn = [System.Net.Dns]::GetHostByName(($env:computerName)).HostName
+    Write-Output "FQDN = $fqdn"
+
+    $OS = [System.Environment]::OSVersion.Platform
+    Write-Output "Operating System is $OS"
+
+    # Create a self-signed certificate
+    if ($OS -eq "Unix") {
+        # Install OpenSSL module
+        Install-Module -Name OpenSSL -Repository PSGallery -Force
+        # Show version of OpenSSL just to make sure it is installed
+        openssl version
+
+        # Create self signed certificate using openssl
+        Write-Output "Creating certificate for linux..."
+        if ($fqdn.length -gt 64) {
+            $machineId = $fqdn.Substring(0,15)
+            openssl req -x509 -newkey rsa:4096 -sha256 -days 1095 -nodes -keyout $OutDir/localhostcert.key -out $OutDir/localhostcert.cer -subj "/CN=$machineId" -addext "subjectAltName=DNS:$fqdn,DNS:localhost,IP:127.0.0.1,IP:::1"
+        }
+        else {
+            openssl req -x509 -newkey rsa:4096 -sha256 -days 1095 -nodes -keyout $OutDir/localhostcert.key -out $OutDir/localhostcert.cer -subj "/CN=$fqdn" -addext "subjectAltName=DNS:$fqdn,DNS:localhost,IP:127.0.0.1,IP:::1"
+        }
+        # Export the certificate to pfx
+        Write-Output "Exporting certificate to pfx..."
+        openssl pkcs12 -export -in $OutDir/localhostcert.cer -inkey $OutDir/localhostcert.key -out $OutDir/localhostcert.pfx -password pass:nopassword
+
+        Write-Output "Converting certificate to pem..."
+        # Create pem from cer
+        cp $OutDir/localhostcert.cer $OutDir/localhostcert.pem
+
+        # Add trust to the pem certificate
+        Write-Output "Adding trust to pem certificate..."
+        openssl x509 -trustout -addtrust "serverAuth" -in $OutDir/localhostcert.pem
+
+        # Import the certificate to the Root store ------------------------------------------------------------------------------
+        # NOTE:  The process must have root privileges to add the certificate to the Root store. If not, then use  
+        #        "chmod 777 /usr/local/share/ca-certificates" to give read, write and execute privileges to anyone on that folder 
+        # Copy the certificate to /usr/local/share/ca-certificates folder while changing the extension to "crt". 
+        # Only certificates with extension "crt" gets added for some reason.
+        Write-Output "Copy the pem certificate to /usr/local/share/ca-certificates folder..."
+        cp $OutDir/localhostcert.pem /usr/local/share/ca-certificates/localhostcert.crt
+
+        # Add trust to the mismatched certificate as well
+        $ openssl x509 -in $OutDir/mismatchedcert.cer -inform der -out $OutDir/mismatchedcert.pem
+        openssl x509 -trustout -addtrust "serverAuth" -in $OutDir/mismatchedcert.pem
+        cp $OutDir/mismatchedcert.pem /usr/local/share/ca-certificates/mismatchedcert.crt
+
+        # Update the certificates store
+        Write-Output "Updating the certificates store..."
+        update-ca-certificates -v
```

Comment:
There are couple of things here, generally it is better to use bash on ubuntu, you will have a better control over the commands. 

1.  Enabling as system CA certificate is missing here. It could be done by [dpkg-reconfigure](https://learn.microsoft.com/en-us/sql/linux/sql-server-linux-encrypted-connections?view=sql-server-ver16&tabs=client#register-the-certificate-on-your-client-machine-windows-linux-or-macos) command
2. Remember since the server and client are on the same machine you need to configure them for both.
3. Removing  '!' at the beginning of certificate inside ca-certificate is missed. You can test with `sudo cat /etc/ca-certificates.conf` command to ensure '!' is removed from the certificate name. If '!' shows up Infront of the certificate name it wont be included in `update-ca-certificate` command.
4. It is better to provide chmod 755/777 to ensure sql server has access to those files.